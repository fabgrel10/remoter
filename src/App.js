import React from 'react';
import './style/main.scss';

function App() {
  return (
    <div className="app">
      <h1>React Capstone Project</h1>
    </div>

  );
}

export default App;
