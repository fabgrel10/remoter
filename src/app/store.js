import { applyMiddleware, createStore } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import jobBoardReducer from '../features/JobBoard/jobBoardReducer';

const store = createStore(jobBoardReducer, applyMiddleware(logger, thunk));

export default store;
