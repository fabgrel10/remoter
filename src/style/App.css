.container {
  display: table;
  width: calc(100vw - 10px);
  margin: 0 auto;
  color: #fff;
  background: #4369b2;
  box-shadow: rgba(100, 100, 111, 0.2) 0 7px 29px 0;
}

.header__section,
.search__section,
.job-board__section {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(390px, 810px));
  margin-bottom: 40px;
  margin-inline: auto;
  width: fit-content;
}

.header__section {
  justify-items: center;
}

.header__content {
  display: flex;
}

.header__text {
  margin-left: 30px;
  font-size: 45px;
}

.header__logo {
  width: 100px;
}

a {
  position: relative;
  font-size: 14px;
  padding: 5px 10px;
  color: white;
  text-transform: uppercase;
  text-decoration: none;
  background-color: #ec4c8a;
  box-shadow: 0 5px 0 0 #8a3153;
}

a:hover {
  background-color: #80324f;
  cursor: pointer;
}

a:active {
  top: 5px;
  box-shadow: none;
}
