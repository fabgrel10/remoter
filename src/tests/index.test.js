import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';
import store from '../redux/store';

afterEach(() => cleanup());

it('index renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
