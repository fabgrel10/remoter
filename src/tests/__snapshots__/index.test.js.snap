// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders correctly 1`] = `
<div
  className="container"
>
  <header
    className="header__section"
  >
    <div
      className="header__content"
    >
      <img
        alt="app logo"
        className="header__logo"
        src="job-offers-bro.png"
      />
      <div>
        <h2
          className="header__text"
        >
          Remoter
        </h2>
      </div>
    </div>
  </header>
</div>
`;
